### PR TITLE
add exp Op

### PR DESCRIPTION
**Pull request by jordan-chalupka**
will add tests

Before:
```
>>> import torch
... from tinygrad import Tensor
... 
... print(torch.tensor([12.0], dtype=torch.bfloat16).exp().tolist())
... print(torch.tensor([12.0], dtype=torch.float32).exp().tolist())
... print(Tensor([12.0], dtype="bfloat16").exp().tolist())
... print(Tensor([12.0], dtype="float32").exp().tolist())
... 
[162816.0]
[162754.796875]
[169984.0]
[162754.703125]
```
After
```
>>> import torch
... from tinygrad import Tensor
... 
... print(torch.tensor([12.0], dtype=torch.bfloat16).exp().tolist())
... print(torch.tensor([12.0], dtype=torch.float32).exp().tolist())
... print(Tensor([12.0], dtype="bfloat16").exp().tolist())
... print(Tensor([12.0], dtype="float32").exp().tolist())
... 
[162816.0]
[162754.796875]
[162816.0]
[162754.796875]
```

This also contributes to inaccuracies in log/cos, see #11756

### CODE REVIEW

**Comment by chenyuxyz** [comment]
Why?

**Comment by jordan-chalupka** [comment]
> Why?

@chenyuxyz looking at a solution for bf16 exp, log, cos inaccuracies 
Gaining context here, is there a reason we don't want this Op?

**Comment by 24f1001789** [comment]
Hey just a question, rather than creating a new uop, wouldnt casting it to least f32 and then after multiplying with `math.log2` for instance, we cast it back to bf32 not do? Im still trying to grasp the codebase... Feel free to ignore if its wrong or irrelevent...

**Comment by geohot** [comment]
Ops are bad, especially new ops in tensor.py. Under the hood what's different here? Look at the GPU assembly, GPUs usually only have EXP2 instructions.

**Comment by jordan-chalupka** [comment]
okay, I'll post a new PR to just do the cast to f32 and back, I checked & the accuracy with that is fine
```
>>> import torch
... from tinygrad import Tensor
... 
... print(torch.tensor([12.0], dtype=torch.bfloat16).exp().tolist())
... print(torch.tensor([12.0], dtype=torch.float32).exp().tolist())
... print(Tensor([12.0], dtype="bfloat16").exp().tolist())
... print(Tensor([12.0], dtype="float32").exp().tolist())
... 
[162816.0]
[162754.796875]
[162816.0]
[162754.703125]
```

**Comment by jordan-chalupka** [comment]
doing this instead https://github.com/tinygrad/tinygrad/pull/11778
